cpp/include/doxygen_groups.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This header is only processed by doxygen and does
+// not need to be included in any source file.
+// Below are the main groups that doxygen uses to build
+// the Modules page in the specified order.
+// Only add a group if it should appear in the root of
+// Modules page's tree.
+
+/**
+ * @defgroup cudf_classes Classes
+ * @{
+ *   @defgroup column_classes Column
+ *   @{
+ *      @defgroup column_factories Factories
+ *      @defgroup strings_classes Strings
+ *      @defgroup dictionary_classes
+ *      @defgroup timestamp_classes Timestamp
+ *   @}
+ *   @defgroup table_classes Table
+ *   @defgroup scalar_classes Scalar
+ *   @{
+ *      @defgroup scalar_factories Factories
+ *   @}
+ * @}
+ * @defgroup column_apis Column and Table APIs
+ * @{
+ *   @defgroup column_copy Copying
```

Comment:
Why are all the group names prefixed with `column_` even when they operate on tables? Should the prefix be dropped?

Reply:
I wanted to have enough of a prefix to be able to search for group-id name but not too long since it may have to be entered manually in comment blocks. Intellisense does not help with doxygen group-id names in comment blocks.

So I don't want `copy` because searching for word `copy` would result in too much noise.

And I don't want `column_and_table_copy` because that seems unnecessarily long.